src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js

```diff
@@ -146,19 +151,17 @@ function FloatingActionButtonAndPopover(props) {
         }
     };
 
-    useEffect(
-        () => {
-            const navigationState = props.navigation.getState();
-            const routes = lodashGet(navigationState, 'routes', []);
-            const currentRoute = routes[navigationState.index];
-            if (currentRoute && ![NAVIGATORS.CENTRAL_PANE_NAVIGATOR, SCREENS.HOME].includes(currentRoute.name)) {
-                return;
-            }
+    useEffect(() => {
+        const navigationState = props.navigation.getState();
+        const routes = lodashGet(navigationState, 'routes', []);
+        const currentRoute = routes[navigationState.index];
+        if (currentRoute && ![NAVIGATORS.CENTRAL_PANE_NAVIGATOR, SCREENS.HOME].includes(currentRoute.name)) {
+            return;
+        }
+        if (!props.showDownloadAppBanner || !Browser.isMobile()) {
```

Comment:
If we're not on a mobile browser why do we ignore the `showDowloadAppBanner` prop and show the welcome menu?

Reply:
Also we prefer early returns when possible.

Reply:
because this only affects mWeb, we will default to the original behavior for all other platforms

Reply:
Ah ok right, because the download app banner is only displayed on mobile web. Please add that explanation in a comment to help future readers of this code.

Reply:
I also would like you to change this to an early return please.